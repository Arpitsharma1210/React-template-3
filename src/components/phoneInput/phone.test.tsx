import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import PhoneInput from './index';
 
jest.mock('react-phone-input-2/lib/bootstrap.css', () => ({}));

describe('PhoneInput Component', () => {
  it('renders without crashing', () => {
    render(<PhoneInput />);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  it('renders label when provided', () => {
    const labelText = 'Phone Number';
    render(<PhoneInput label={labelText} />);
    expect(screen.getByText(labelText)).toBeInTheDocument();
  });

  it('renders error message when provided', () => {
    const errorMessage = 'Invalid phone number';
    render(<PhoneInput error={errorMessage} />);
    expect(screen.getByText(errorMessage)).toBeInTheDocument();
  });

//   it('displays read-only text when in read-only mode', () => {
//     const readOnlyText = '+1  123-456-7890';
//     render(<PhoneInput value={{ phone: readOnlyText }} readOnly />);
//     expect(screen.getByDisplayValue(readOnlyText)).toBeInTheDocument();
//   });

  it('changes phone input value when a new value is entered', () => {
    const onChangeMock = jest.fn();
    render(<PhoneInput onChange={onChangeMock} />);
    const phoneInput = screen.getByRole('textbox');
    fireEvent.change(phoneInput, { target: { value: '+1  123-456-7890' } });
    expect(onChangeMock).toHaveBeenCalledWith({ phone: '11234567890', dialCode: '1' });
  });

//   it('takes up full width when fullWidth prop is true', () => {
//     render(<PhoneInput fullWidth />);
//     const phoneInput = screen.getByRole('textbox');
//     expect(phoneInput).toHaveStyle('max-width:  100%');
//   });

//   it('shows required indicator when required prop is true', () => {
//     render(<PhoneInput required />);
//     const phoneInput = screen.getByRole('textbox');
//     expect(phoneInput).toHaveAttribute('required');
//   });

  it('is disabled when disabled prop is true', () => {
    render(<PhoneInput disabled />);
    const phoneInput = screen.getByRole('textbox');
    expect(phoneInput).toBeDisabled();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<PhoneInput />);
    expect(asFragment()).toMatchSnapshot();
  });
});
