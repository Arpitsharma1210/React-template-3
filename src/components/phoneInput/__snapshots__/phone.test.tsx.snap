// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneInput Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcHOTY jMBOyb"
  >
    <div
      class="sc-dkHyXG fKEDZB"
    >
      <div
        class=" react-tel-input sc-iBIYEh gaaOUi"
      >
        <div
          class="special-label"
        >
          Phone
        </div>
        <input
          class="form-control "
          placeholder="1 (702) 123-4567"
          type="tel"
          value="+1"
        />
        <div
          class="flag-dropdown "
        >
          <div
            aria-haspopup="listbox"
            class="selected-flag"
            role="button"
            tabindex="0"
            title="United States: + 1"
          >
            <div
              class="flag us"
            >
              <div
                class="arrow"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 sc-jTkTEj bXYaQ css-e784if-MuiTypography-root"
    />
  </div>
</DocumentFragment>
`;
